### PR TITLE
Introduction of gcc-11 to CI 

### PR DESCRIPTION
**Pull request by pwisniewskimobica**
Switch to gcc11 and g++11

### CODE REVIEW

**Comment by pwisniewskimobica** [comment]
We want to introduce compilation with gcc-11 in Flang CI.
This PR lets us build the gcc-11 artifacts for later use in Flang repository CI which is done in
flang-compiler/flang#1190

Kind request for your reviews, @bryanpkc , @shivaramaarao , @mleair , @kiranchandramohan
Thank you in advance!

**Review by bryanpkc — APPROVED** [review]
LGTM

**Comment by bryanpkc** [comment]
I would suggest rewording the commit to something like:
```
Use GCC 11 in CI builds
```

File: .github/workflows/pre-compile_llvm.yml [hunk 1]
```diff
@@ -3,26 +3,21 @@ name: Pre-compile llvm
 on:
   workflow_dispatch:
   push:
-    branches: [ release_100, release_11x, release_12x ]
+    branches: [ release_100, release_11x, release_12x, release_13x]
```
Comment (pawosm-arm) [inline .github/workflows/pre-compile_llvm.yml]: As @bryanpkc said before, please keep spaces where they were...

**Review by pawosm-arm — CHANGES_REQUESTED** [review]

File: .github/workflows/pre-compile_llvm.yml (hunk 1 above)
Comment (pwisniewskimobica) [reply .github/workflows/pre-compile_llvm.yml]: -->fixed

**Review by RichBarton-Arm — APPROVED** [review]
Happy to approve from Arm.

@shivaramaarao - happy to put this on 12 also?

**Comment by RichBarton-Arm** [comment]
As per the policy, only one approval is needed on CI changes, so I am just going to merge this.

**Comment by bryanpkc** [comment]
@RichBarton-Arm  For future reference, please avoid merge commits. They make it hard to see what happened in one-line git logs.
